File: Languages/JavaScript/Day024.js
// Languages - JavaScript - Day 24
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that lists all Prime Numbers up to a given number n.


function isPrime(num) {
	if (num < 2) return false;
	if (num === 2 || num === 3) return true;
	if (num % 2 === 0 || num % 3 === 0) return false;
	let i = 5;
	while (i * i <= num) {
		if (num % i === 0 || num % (i + 2) === 0) return false;
		i += 6;
	}
	return true;
}

function listPrimes(n) {
	if (n < 2) {
		return [];
	}
	return Array.from({ length: n - 1 }, (_, i) => i + 2).filter(isPrime);
}

const input = prompt("Enter a number:").trim();

if (input === "") {
	console.log("Invalid input. Input cannot be empty.");
} else {
	const n = parseInt(input, 10);
	if (Number.isInteger(n) && n >= 0) {
		const primes = listPrimes(n);
		if (primes.length > 0) {
			console.log(`Prime numbers up to ${n}: ${primes.join(", ")}`);
		} else {
			console.log(`No prime numbers found up to ${n}.`);
		}
	} else {
		console.log("Invalid input. Please enter a valid non-negative integer.");
	}
}

File: Languages/R/Day024.r
# Languages - R - Day 24
# Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

# Create a function that lists all Prime Numbers up to a given number n.


is_prime <- function(num) {
  if (num <= 1) return(FALSE)
  if (num == 2 || num == 3) return(TRUE)
  if (num %% 2 == 0 || num %% 3 == 0) return(FALSE)
  i <- 5
  while (i * i <= num) {
    if (num %% i == 0 || num %% (i + 2) == 0) return(FALSE)
    i <- i + 6
  }
  return(TRUE)
}

list_primes <- function(n) {
  if (n < 0) {
    cat("Please enter a non-negative number.\n")
    return(integer(0))
  }
  if (n < 2) {
    return(integer(0))
  }
  return(Filter(is_prime, 2:n))
}

cat("Enter a number: ")
input <- suppressWarnings(as.integer(readLines("stdin", n = 1)))

if (!is.na(input) && input >= 0) {
  primes <- list_primes(input)
  if (length(primes) == 0) {
    cat("No prime numbers found up to", input, "\n")
  } else {
    cat("Prime numbers up to", input, ":", paste(primes, collapse = ", "), "\n")
  }
} else {
  cat("Invalid input. Please enter a valid non-negative number.\n")
}
